BIC_codes/functions/dFC_funcs.py

```diff
@@ -0,0 +1,3472 @@
+    for i in range(F.shape[0]):
+        K = np.zeros((N, N))
+        K[iu] = F[i,:]
+        K = K + K.T
+        K = K + np.eye(N)
+        C.append(K)
+    C = np.array(C)
+    return C
+
+# test
+def common_subj_lst(time_series_dict):
+    SUBJECTs = None
+    for session in time_series_dict:
+        if SUBJECTs is None:
+            SUBJECTs = time_series_dict[session].subj_id_lst
+        else:
+            SUBJECTs = intersection(SUBJECTs, time_series_dict[session].subj_id_lst)
+    return SUBJECTs
+
+def intersection(lst1, lst2): # input is a list 
+    lst3 = [value for value in lst1 if value in lst2]
+    return lst3
+
+def TR_intersection(dFCM_lst): # input is a list of dFCM objs
+    TRs_lst_old = dFCM_lst[0].TR_array
+    for dFCM in dFCM_lst:
+        TRs_lst_new = intersection(TRs_lst_old, dFCM.TR_array)
+        TRs_lst_old = TRs_lst_new
+    TRs_lst_old.sort()
+    if len(TRs_lst_old)==0:
+        print('No TR intersection.')
+    return TRs_lst_old
+
+def dFC_dict_slice(data, idx_lst):
+    data_sliced = {}
+    for i, k in enumerate(data):
+        if i in idx_lst:
+            data_sliced[k] = data[k]
+    return data_sliced
+
+def visualize_state_TC(TC_lst, \
+    TRs, \
+    state_lst, \
+    TC_name_lst, \
+    title='', \
+    save_image=None, output_root=None\
+    ):
+
+    color_lst = ['k', 'b', 'g', 'r']
+
+    if 'on' in state_lst and 'off' in state_lst:
+        ticks = range(2)
+    else:
+        ticks = range(1, len(state_lst)+1)
+
+    plt.figure(figsize=(25, 5))
+    for i, TC in enumerate(TC_lst):
+        plt.plot(TRs, TC, color_lst[i], linewidth=2)
+    plt.xlabel('TR')
+    plt.yticks(ticks=ticks, labels=state_lst)
+    plt.legend(TC_name_lst)
+    plt.title(title)
+    if save_image:
+        folder = output_root[:output_root.rfind('/')]
+        if not os.path.exists(folder):
+            os.makedirs(folder)
+        plt.savefig(output_root+'.png', \
+            dpi=fig_dpi, bbox_inches=fig_bbox_inches, pad_inches=fig_pad \
+        ) 
+        plt.close()
+    else:
+        plt.show()
+
+    return
+
+def visualize_conn_mat(data, title='', \
+    name_lst_key=None, mat_key=None, \
+    cmap='viridis',\
+    normalize=False,\
+    disp_diag=True,\
+    save_image=False, output_root=None, \
+        fix_lim=True, lim_val=1.0 \
+    ):
+
+    '''
+    - name_lst_key can be a list of names or the key to list of names
+    - data must be a dict of correlation/connectivity matrices
+    sample:
+    Suptitle1
+        corr_mat
+            0.00 0.31 0.76 
+            0.31 0.00 0.43 
+            0.76 0.43 0.00 
+        measure_lst
+            ContinuousHMM
+            Windowless
+            Clustering_pear_corr
+    Suptitle1
+        corr_mat
+            0.00 0.32 0.76 
+            0.32 0.00 0.45 
+            0.76 0.45 0.00 
+        measure_lst
+            ContinuousHMM
+            Windowless
+            Clustering_pear_corr
+    '''
+
+    if name_lst_key is None:
```

Comment:
try to keep function short; they should at least fit on a screen ie less than ~40 lines (there may be  rare exceptions), and in most cases be even much shorter than that (~5-10 lines)